code/modules/mob/living/living.dm

```diff
@@ -426,6 +426,27 @@
 
 		set_pull_offsets(M, state)
 
+/// Adds pass flags and signal to smash through windows as you fly through the air
+/mob/living/proc/start_window_flight(duration = 1.5 SECONDS)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(flying_window_smash))
+	passwindow_on(src, TRAIT_MOVE_FLYING)
```

Comment:
`TRAIT_MOVE_FLYING` is a trait used for movetypes, not the best of the sources really.

Reply:
Ah, was wondering. I'll find something else.